src/main/java/gregtech/common/metatileentities/multi/multiblockpart/appeng/MetaTileEntityAEHostablePart.java

```diff
@@ -198,4 +204,41 @@ protected IMEMonitor<T> getMonitor() {
             return null;
         }
     }
+
+    public EnumSet<EnumFacing> getConnectableSides() {
+        return this.allowExtraConnections ? EnumSet.allOf(EnumFacing.class) : EnumSet.of(getFrontFacing());
+    }
+
+    public void updateConnectableSides() {
+        if (this.aeProxy != null) {
+            this.aeProxy.setValidSides(getConnectableSides());
+        }
+    }
+
+    @Override
+    public boolean onWireCutterClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing,
+                                     CuboidRayTraceResult hitResult) {
+        this.allowExtraConnections = !this.allowExtraConnections;
+        updateConnectableSides();
+
+        if (!getWorld().isRemote) {
+            playerIn.sendStatusMessage(new TextComponentTranslation(this.allowExtraConnections ?
+                    "gregtech.machine.me.extra_connections.enabled" : "gregtech.machine.me.extra_connections.disabled"),
+                    true);
+        }
+
+        return true;
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound data) {
+        data.setBoolean("AllowExtraConnections", this.allowExtraConnections);
+        return super.writeToNBT(data);
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound data) {
+        super.readFromNBT(data);
+        this.allowExtraConnections = data.getBoolean("AllowExtraConnections");
```

Comment:
reading/writing NBT is very order dependent. You always want to read/write in the same order. Here you are writing the extra connections first, and then super, but reading super and then extra connections.

The order should match, most likely with super written first.